test/extended/builds/build_pruning.go

```diff
@@ -93,4 +91,32 @@ var _ = g.Describe("[builds][pruning] prune builds based on settings in the buil
 		o.Expect(int32(len(builds.Items))).To(o.Equal(*buildConfig.Spec.FailedBuildsHistoryLimit), "there should be %v failed builds left after pruning, but instead there were %v", *buildConfig.Spec.FailedBuildsHistoryLimit, len(builds.Items))
 
 	})
+
+	g.It("should prune canceled builds based on the failedBuildsHistoryLimit setting", func() {
```

Comment:
you're using the successfulBuildConfig fixture, but that buildconfig doesn't have a failedBuildsHistoryLimit...